packages/jest-mock/src/index.ts

```diff
@@ -988,6 +1003,10 @@ export class ModuleMocker {
     return metadata;
   }
 
+  isMockFunction<T, Y extends Array<unknown> = Array<unknown>>(
+    fn: (...args: Y) => T,
+  ): fn is Mock<T, Y>;
+  isMockFunction(fn: unknown): boolean;
```

Comment:
The first overload will infer return type and types of args if a function is passed. Second one allows to pass any value. This way types do the same job as before, but with added feature.

Reply:
should second be `isMockFunction(fn: unknown): fn is Mock<unknown, unknown>` instead?

Reply:
In this case, if `string` is passed the typed in `if (isMockFunction(someString))` looks like this:

<img width="658" alt="Screenshot 2022-02-21 at 07 39 48" src="https://user-images.githubusercontent.com/72159681/154895980-b021d984-5806-4425-b00f-cba723249b58.png">

No harm at all, because this branch will not execute anyway. Simply seemed unnecessary. So `boolean` looked like cleaner solution.

Reply:
what if `hardlyMock` is `any` or `unknown`?

Reply:
These are good edge cases. I added tests

Reply:
right, and then `isMockFunction(fn: unknown): fn is Mock<unknown, unknown>` still doesn't help?